pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -235,7 +281,21 @@ func (*Plugin) OnInboundListener(in *plugin.CallbackListenerInputParams, mutable
 		// Only care about Router nodes.
 		return nil
 	}
-	// TODO: implementation
+	authnPolicy := model.GetConsolidateAuthenticationPolicy(
+		in.Env.Mesh, in.Env.IstioConfigStore, in.ServiceInstance.Service.Hostname, in.ServiceInstance.Endpoint.ServicePort)
+
+	if mutable.Listener == nil || len(mutable.Listener.FilterChains) != 1 {
+		return fmt.Errorf("Expect lister has exactly one filter chain")
```

Comment:
`fmt.Errorf("expect listener %v to have exactly one filter change", mutable.listener)`

And make this a warning at most; Error is for stuff that crashes the server.

Reply:
This is to return error from this function. Do you want just log warning and return nil ?

Reply:
Sorry, I conflated the two:I'm so used to having to tell people not to log at `log.Error` I fell into it out of habit. :P No just return the `fmt.Errorf`, but per golang style error strings should be lower case.